receiver/kubeletstatsreceiver/internal/kubelet/accumulator.go

```diff
@@ -49,66 +49,49 @@ type metricDataAccumulator struct {
 	logger                *zap.Logger
 	metricGroupsToCollect map[MetricGroup]bool
 	time                  time.Time
+	mbs                   *metadata.MetricsBuilders
 }
 
-const (
-	scopeName = "otelcol/kubeletstatsreceiver"
-)
-
 func (a *metricDataAccumulator) nodeStats(s stats.NodeStats) {
 	if !a.metricGroupsToCollect[NodeMetricGroup] {
 		return
 	}
-
-	md := pmetric.NewMetrics()
-	rm := md.ResourceMetrics().AppendEmpty()
-	fillNodeResource(rm.Resource(), s)
-
-	ilm := rm.ScopeMetrics().AppendEmpty()
-	ilm.Scope().SetName(scopeName)
-
-	startTime := pcommon.NewTimestampFromTime(s.StartTime.Time)
+	a.mbs.WithNodeStartTime.Reset(metadata.WithStartTime(pcommon.NewTimestampFromTime(s.StartTime.Time)))
```

Comment:
This is not the right use case of `Reset`. `metadata.WithStartTime` should be passed to the initializer during receiver start time only once. We probably need to run `r.statsProvider.StatsSummary()` additionally during the receiver start time

Reply:
I don't think that will work for Pods, since right now StartTime is Pod specific.  We'd need a metrics builder per pod if we can't reset start time.  

It's possible that the actually StartTime value returned in the summary for each pod is the same, but I don't think we can definitively say that.

Reply:
I see what you mean. We have different start time per resource metrics. Let me submit a change to the metrics builder to unblock this

Reply:
Submitted this https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9992

Reply:
@dmitryax updated to use the new start time override feature.  Also, reviewing the existing code I noticed container's use their own start time, not Pod's start time, so I added a fourth metrics builder for containers.